src/processor/handler.rs

```diff
@@ -163,253 +151,32 @@ impl From<ReceiveError> for ContextError {
-                )))
-            }
-            TpStateSetResponse_Status::STATUS_UNSET => Err(ContextError::ResponseAttributeError(
-                String::from("Status was not set for TpStateSetResponse"),
-            )),
-        }
-    }
+pub trait TransactionContext {
+    #[deprecated(
+        since = "0.3.0",
+        note = "please use `get_state_entry`  or  `get_state_entries` instead"
+    )]
+    fn get_state(&self, addresses: &[String]) -> Result<Vec<(String, Vec<u8>)>, ContextError>;
 
-    /// delete_state requests that each of the provided addresses be unset
-    /// in validator state. A list of successfully deleted addresses
-    ///  is returned.
-    ///
-    /// # Arguments
-    ///
-    /// * `addresses` - the addresses to fetch
-    #[allow(clippy::needless_pass_by_value)]
-    pub fn delete_state(
-        &mut self,
-        addresses: Vec<String>,
-    ) -> Result<Option<Vec<String>>, ContextError> {
-        let mut request = TpStateDeleteRequest::new();
-        request.set_context_id(self.context_id.clone());
-        request.set_addresses(RepeatedField::from_vec(addresses.clone()));
+    fn get_state_entry(&self, addresses: &str) -> Result<Option<Vec<u8>>, ContextError>;
 
-        let serialized = request.write_to_bytes()?;
-        let x: &[u8] = &serialized;
-
-        let mut future = self.sender.send(
-            Message_MessageType::TP_STATE_DELETE_REQUEST,
-            &generate_correlation_id(),
-            x,
-        )?;
-
-        let response: TpStateDeleteResponse =
-            protobuf::parse_from_bytes(future.get()?.get_content())?;
-        match response.get_status() {
-            TpStateDeleteResponse_Status::OK => Ok(Some(Vec::from(response.get_addresses()))),
-            TpStateDeleteResponse_Status::AUTHORIZATION_ERROR => {
-                Err(ContextError::AuthorizationError(format!(
-                    "Tried to delete unauthorized address: {:?}",
-                    addresses
-                )))
-            }
-            TpStateDeleteResponse_Status::STATUS_UNSET => {
-                Err(ContextError::ResponseAttributeError(String::from(
-                    "Status was not set for TpStateDeleteResponse",
-                )))
-            }
-        }
-    }
-
-    /// add_receipt_data adds a blob to the execution result for this transaction
-    ///
-    /// # Arguments
-    ///
-    /// * `data` - the data to add
-    pub fn add_receipt_data(&mut self, data: &[u8]) -> Result<(), ContextError> {
-        let mut request = TpReceiptAddDataRequest::new();
-        request.set_context_id(self.context_id.clone());
-        request.set_data(Vec::from(data));
+    fn get_state_entries(
+        &self,
+        addresses: &[String],
+    ) -> Result<Vec<(String, Vec<u8>)>, ContextError>;
 
-        let serialized = request.write_to_bytes()?;
-        let x: &[u8] = &serialized;
+    fn set_state(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError>;
```

Comment:
set_state and delete_state should mirror what we are doing w/get_state_entry/entries. In addition to the naming, get_state_entries uses a Vec and not HashMap.